src/Plugin/PrefetchIterator.php

```diff
@@ -58,7 +58,7 @@ class PrefetchIterator extends AbstractPlugin implements \Iterator, \Countable
      *
      * @var int
      */
-    protected $position;
+    protected $position = 0;
```

Comment:
isn't defining the default in the constructor sufficient?

Reply:
The example in the PHP Manual does both:
```php
<?php
class myIterator implements Iterator {
    private $position = 0;
    // ...  

    public function __construct() {
        $this->position = 0;
    }
```

As it is now, none of the plugin classes have an explicit constructor defined. The parent constructor — `Configurable::__construct()` — is called from `AbstractPlugin::__construct()`.

Following the logic of `Configurable`, this should be done in `PrefetchIterator::init()`? But then we might as well rename `resetData()` to `init()` …

Reply:
ah yes, i see...

personally i think ``resetData`` is fine